airflow-core/src/airflow/api_fastapi/execution_api/routes/xcoms.py

```diff
@@ -240,6 +247,8 @@ def get_mapped_xcom_by_slice(
     params: Annotated[GetXComSliceFilterParams, Query()],
     session: SessionDep,
 ) -> XComSequenceSliceResponse:
+    if key:
+        key = unquote(key)
```

Comment:
Do we really need the unqoute? I would have expected FastAPI to unquote it before it gives it to us.